eureka/S4_generate_lightcurves/s4_genLC.py

```diff
@@ -144,91 +149,122 @@ def lcJWST(eventlabel, ecf_path='./', s3_meta=None):
+                                        np.ma.MaskedArray.count(opterr_ma)
 
                 # Do 1D sigma clipping (along time axis) on binned spectra
                 if meta.sigma_clip:
-                    meta.lcdata[i], outliers = clipping.clip_outliers(meta.lcdata[i], log, wave_1d[l], meta.sigma, meta.box_width, meta.maxiters, meta.boundary, meta.fill_value, verbose=False)
-                    log.writelog('  Sigma clipped {} outliers in time series'.format(outliers))
+                    lc['data'][i], outliers = clipping.clip_outliers(lc['data'][i].values, log, spec.wave_1d[i], meta.sigma, meta.box_width, meta.maxiters, meta.boundary, meta.fill_value, verbose=False)
+                    log.writelog('  Sigma clipped {} outliers in time series'.format(outliers), mute=(not meta.verbose))
 
                 # Plot each spectroscopic light curve
                 if meta.isplots_S4 >= 3:
-                    plots_s4.binned_lightcurve(meta, meta.time, i)
+                    plots_s4.binned_lightcurve(meta, lc, i)
 
             # Calculate total time
             total = (time_pkg.time() - t0) / 60.
             log.writelog('\nTotal time (min): ' + str(np.round(total, 2)))
 
-            log.writelog('Saving results as astropy table')
+            log.writelog('Saving results')
             event_ap_bg = meta.eventlabel + "_ap" + str(spec_hw_val) + '_bg' + str(bg_hw_val)
-            meta.tab_filename_s4 = meta.outputdir + 'S4_' + event_ap_bg + "_Table_Save.txt"
-            wavelengths = np.mean(np.append(meta.wave_low.reshape(1,-1), meta.wave_hi.reshape(1,-1), axis=0), axis=0)
-            wave_errs = (meta.wave_hi-meta.wave_low)/2
-            astropytable.savetable_S4(meta.tab_filename_s4, meta.time, wavelengths, wave_errs, meta.lcdata, meta.lcerr)
+            # Save Dataset object containing time-series of 1D spectra
+            meta.filename_S4_SpecData = meta.outputdir + 'S4_' + event_ap_bg + "_SpecData.h5"
+            success = xrio.writeXR(meta.filename_S4_SpecData, spec, verbose=True)
+            # Save Dataset object containing binned light curves
+            meta.filename_S4_LCData = meta.outputdir + 'S4_' + event_ap_bg + "_LCData.h5"
+            success = xrio.writeXR(meta.filename_S4_LCData, lc, verbose=True)
+
+            # wavelengths = np.mean(np.append(meta.wave_low.reshape(1,-1), meta.wave_hi.reshape(1,-1), axis=0), axis=0)
+            # wave_errs = (meta.wave_hi-meta.wave_low)/2
+            # astropytable.savetable_S4(meta.tab_filename_s4, meta.time, wavelengths, wave_errs, meta.lcdata, meta.lcerr)
 
             # Save results
-            log.writelog('Saving results')
             me.saveevent(meta, meta.outputdir + 'S4_' + meta.eventlabel + "_Meta_Save", save=[])
 
-            # if (isplots >= 1) and (correctDrift == True):
-            #     # Plot Drift
-            #     # Plots corrected 2D light curves
-
             log.closelog()
 
-    return meta
+    return spec, lc, meta
```

Comment:
If we don't end up chaging spec in this stage (given my two comments above), then we don't need to return it here